app/sources/HFTemplateController.m

```diff
@@ -380,6 +380,26 @@ - (BOOL)readMacDate:(NSDate **)value forLabel:(NSString *)label {
     return YES;
 }
 
+- (NSString *)readFatDateWithLabel:(NSString *)label error:(NSString **)error {
+    int16_t val;
+    if (![self readInt16:&val forLabel:nil]) {
+        if (error) {
+            *error = @"Failed to read int16 bytes";
+        }
+        return nil;
+    }
+
+    NSInteger day = val & 0x1F;
+    NSInteger month = val >> 5 & 0xF;
+    NSInteger year = 1980 + ((val >> 9) & 0x7F);
+    NSString *date = [NSString stringWithFormat:@"%d-%02d-%02d", year, month, day];
```

Comment:
This line is generating a warning:

> HFTemplateController.m:395:79: Values of type 'NSInteger' should not be used as format arguments; add an explicit cast to 'long' instead

Let's just use `int` instead of `NSInteger` to keep it simpler.